tests/publiccloud/check_registercloudguest.pm

```diff
@@ -67,6 +66,9 @@ sub run {
             die('/var/log/cloudregister is not empty');
         }
         $instance->ssh_assert_script_run(cmd => '! sudo SUSEConnect -d', fail_message => 'SUSEConnect succeeds but it is not supported should fail on BYOS');
+    } elsif (is_byos()) {
```

Comment:
WDYT if we will move this piece inside `elsif (is_byos())` in line 47 and then within this elsif there will be another `if(!check_var('PUBLIC_CLOUD_CHECK_CLOUDREGISTER_EXECUTED', '1'))`

I mean that for me personally between : 
```
elsif(a_codition && b_condition) {
code1
} elif ( a_condition ) {
code2
}
```

AND

```
elsif(a_codition) {
if (b_condition){
code1
}
else {
code2
}
}
```

I would certainly prefer second setup

Reply:
I tried to avoid some 🍝 code

Reply:
so you saying that first version is NOT spagetti code and second one is ?

Reply:
ok

Reply:
It's 🍝 time, I've changed my mind